### ISSUE TITLE
Strange behaviour when linting blocks.

### ISSUE THREAD

**Issue by g1ps** [ansible-community/ansible-lint#671]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version  2.8.2
ansible-lint --version 4.2.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

No 'errors' should be reported. Blocks should be linted correctly.

# Actual Behaviour (Bug report only)

I have a file which has two blocks containing two tasks each:

```
- name: ...
  block:
...
 when: ...
```
plus a couple of naked tasks for a total of 34 lines. Linting spits out a couple of dozen of these lines:

```
Error trying to append skipped rules: RuntimeError('Unexpected file type: block',)
```
but continues and passes all files. I have no idea what it thinks it's doing or whether the code is actually being checked properly. There are no lint annotations whatsoever in this file.

**Comment by hispanico** [comment]
I got the same error using ansible-lint v4.2.0 when I going to check tasks in a ansible-role that contain `include_tasks` in a `block`.
```
- name: Somthing
  block:
  - name: Include task
    include_tasks: check.yml
  when: check is defined
```

I've downgrade to ansible-lint v4.1.0 and I don't have that error.

**Comment by webknjaz** [comment]
cc @ssbarnea

**Comment by ssbarnea** [comment]
Closing as we also have a test that tests exactly this.
